#! /usr/bin/env python

PACKAGE='kcl_sbc'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("f_ref", double_t,    0,    "Force Reference", 0, 0.0,	5)
gen.add("beta", double_t,    0,    "Safety Margin", 0.35, 0.0,	3)
gen.add("delta",       double_t,    0,    "Control effort", 0.02,     0, 1)
gen.add("eta",       double_t,    0,    "Step Size", 0.0000001,     0.0000001, 0.001)
gen.add("multiplier",       double_t,    0,    "Speed Multiplier", 1000,     1, 10000)

#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                       gen.const("Medium",     int_t, 1, "A medium constant"),
#                       gen.const("Large",      int_t, 2, "A large constant"),
#                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                     "An enum to set size")
#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "kcl_sbc", "SBC_params"))
